### PR TITLE
Sudoku example fix

### PR DESCRIPTION
**Pull request by Turupawn**
**The problem**

the logic behind `countDuplicates` is invalid because it's doing this:

(incorrect)
```
if(duplicates + e11 == e21)
  duplicates = 1;
else
  duplicates = 0;
```

Instead of this:
(correct)
```
if(e11 == e21)
  duplicates = duplicates + 1;
else
  duplicates = duplicates;
```

**The solution**

I'm using an auxiliary variable to fix this

**Alternative solution**

We can also fix it with an array and a loop. But I went for the auxiliary variable for no particular reason.

```
def countDuplicates(field e11, field e12, field e21, field e22) -> u32 {
    u32[6] mut duplicates = [0,0,0,0,0,0];
    duplicates[0] = e11 == e12 ? 1 : 0;
    duplicates[1] = e11 == e21 ? 1 : 0;
    duplicates[2] = e11 == e22 ? 1 : 0;
    duplicates[3] = e12 == e21 ? 1 : 0;
    duplicates[4] = e12 == e22 ? 1 : 0;
    duplicates[5] = e21 == e22 ? 1 : 0;

    u32 mut count = 0;
    for u32 i in 0..5 {
        count = count + duplicates[i];
    }

    return count;
}
```

And btw, I also added an assert to make sure we check for this validation. I've tested this on the playground and on chain on goerli and with this changes it works correctly.

Thanks for putting adding the examples. It has helped me a lot.
Cheers!

### CODE REVIEW

**Comment by Schaeff** [comment]
Oh good catch, seems like a precedence issue, would a smaller change like the following work?

```diff
-   duplicates = duplicates + e11 == e21 ? 1 : 0;
+   duplicates = duplicates + (e11 == e21 ? 1 : 0);
```

File: zokrates_cli/examples/sudoku/sudoku_checker.zok [hunk 1]
```diff
@@ -74,5 +81,6 @@ def main(field a21, field b11, field b22, field c11, field c22, field d21, priva
     duplicates = duplicates + countDuplicates(b12, b22, d12, d22);
 
     // the solution is correct if and only if there are no duplicates
+    assert(duplicates == 0);
```
Comment (Schaeff) [inline zokrates_cli/examples/sudoku/sudoku_checker.zok]: This is fine but then let's return nothing

**Comment by Turupawn** [comment]
The smaller change worked! I tested it on the playground and works well :+1: 
I'm also returning nothing on the main function.
Thanks for reviewing this @Schaeff !

**Comment by Turupawn** [comment]
Woops, closed this by accident. Just reopened

**Comment by Turupawn** [comment]
After further thought, I went back returning a bool instead of nothing. I think it makes more sense. Specially for newcomers like myself.
So in this PR I'm only fixing the precedence issue.

**Comment by Turupawn** [comment]
Hey guys, I wanted to know if you guys think something else should be done on this PR. I would greatly appreciate if we merge this because I'm doing educational content and it would be nice to point to this official repo instead of my own.
This is the first piece of documentation for reference, but I plan on doing more:
https://dev.to/turupawn/circuitos-zksnarks-paso-a-paso-4i6n
It rewards the players with tokens when they submit a valid proof that they solved the sudoku.
Cheers!

**Comment by Schaeff** [comment]
Yes we can merge it soon. Can you please just add a changelog at `changelogs/unreleased/1287-Turupawn`? Thanks!

**Comment by Turupawn** [comment]
Added the changelog @Schaeff 
Very appreciated :pray:

**Review by Schaeff — APPROVED** [review]

**Pull request merged**